src/controllers/authentication.controller.js

```diff
@@ -0,0 +1,86 @@
+const Joi = require("joi")
+const jwt = require("jsonwebtoken")
+const bcrypt = require("bcrypt")
+const Doctors = require("../models/doctor.model");
+const Patients = require("../models/patient.model")
+
+const {authSchema} = require("../validators/auth.validate");
+
+
+var refreshTokens = []
+
+exports.login = async (req,res) => {
+    try{
+        const result = await authSchema.validateAsync(req.body)
+        
+
+        let doctor = await Doctors.findOne({doctorEmail :result.email})
+        let patient = await Patients.findOne({patientEmail :result.email})
+        
+        if(doctor != null ){
+        if (await bcrypt.compare(result.password, doctor.doctorPassword)) 
+            {   
+                
+                
+                //access token for buyer
+                const accessToken = jwt.sign({doctor}, process.env.DOCTOR_ACCESS_TOKEN_SECRET, {expiresIn: "60m"})
+                //refresh token for seller
+                const refreshToken = jwt.sign({doctor}, process.env.DOCTOR_REFRESH_TOKEN_SECRET, {expiresIn: "20m"})
+                refreshTokens.push(refreshToken)
+                return res.json (
+                    {
+                        status : "Success",
+                        error : false,
+                        message : "Doctor successfully logged in ",
+                        data : {accessToken: accessToken, refreshToken: refreshToken }
+                    }
```

Comment:
I suggest you have the token generation method in a file namely `common.js` inside the `src/utils` folder as this is a common function so it would be better to reuse the function code.

Reply:
Added [here](https://github.com/Sankalp-Sachapara/HospitalPractice/pull/3/files#r1045549120)

Reply:
👍